packit_service/worker/helpers/build/build_helper.py

```diff
@@ -766,9 +768,25 @@ def notify_about_failure_if_configured(self):
         ):
             return
 
-        formatted_message = configured_message.format(
-            commit_sha=self.db_project_event.commit_sha
-        )
+        loop_count = 0
+        new_kwargs = copy.deepcopy(kwargs)
+        formatted_message = None
+        while formatted_message is None and loop_count <= len(
+            ERROR_COMMENT_MESSAGE_DETAILS
+        ):
+            # every detail (the user would like to have)
+            # could be missing in the calling job
+            # iterate over all the details until we can build the user string
```

Comment:
Since you have a name of all the keys, can't we specify all and the unknown to `None`/`"unknown"`?

Reply:
But I like you are trying to resolve this situation.

Also, it would be nice to cover this in the tests...

Reply:
I think I am not following you. There exist a test for the situation, I wrote the code using the test. Or I don't understand which situation do you mean 😅.
Regarding the first hint, I thought that a message like `copr build logs {no entry for copr_build_packit_info} ` can be more understandable for the user than `copr build logs None/unknown` but maybe I did not understand your suggestion?

Reply:
I mixed multiple points here...;)

Tests:
* My bad, I missed that test-case. Sorry.

Wording:
* I agree `{no entry for copr_build_packit_info}` is better than `None/unknown`. => I like the UX.

Code:
* That was my main point if we can't do this in a "less technical" way. (Or, how to describe it.)  Since we have the list of all the possible variables defined in `constants.py`, we can always provide all the variables to the formatted (with `no entry ...` as default) and avoid catching of the `KeyError`.

Reply:
Thanks, now I understand and I agree.